test/test_grudge.py

```diff
@@ -1138,6 +1138,88 @@ def test_empty_boundary(actx_factory):
         assert len(component) == len(discr.discr_from_dd(BTAG_NONE).groups)
 
 
+def test_operator_compiler_overwrite(actx_factory):
+    """Tests that the same expression in ``eval_code`` and ``discr_code``
+    does not confuse the OperatorCompiler in grudge/symbolic/compiler.py.
+    """
+
+    actx = actx_factory()
+
+    ambient_dim = 2
+    target_order = 4
+
+    from meshmode.mesh.generation import generate_regular_rect_mesh
+    mesh = generate_regular_rect_mesh(
+            a=(-0.5,)*ambient_dim, b=(0.5,)*ambient_dim,
+            n=(8,)*ambient_dim, order=1)
+    discr = DiscretizationCollection(actx, mesh, order=target_order)
+
+    # {{{ test
+
+    sym_u = sym.nodes(ambient_dim)
+    sym_div_u = sum(d(u) for d, u in zip(sym.nabla(ambient_dim), sym_u))
+
+    div_u = bind(discr, sym_div_u)(actx)
+    error = bind(discr, sym.norm(2, sym.var("x")))(actx, x=div_u - discr.dim)
+    logger.info("error: %.5e", error)
+
+    # }}}
+
+
+@pytest.mark.parametrize("ambient_dim", [
+    2,
+    pytest.param(3, marks=pytest.mark.xfail)
```

Comment:
This still fails due to some bug in the greedy aggregation.

Reply:
```suggestion
    # FIXME, cf. https://github.com/inducer/grudge/pull/78/
    pytest.param(3, marks=pytest.mark.xfail)
```